src/test/java/org/cactoos/experimental/ThreadsTest.java

```diff
@@ -88,9 +92,130 @@ public void oldSchoolWay() {
                     extor.shutdownNow();
                 }
             } catch (final InterruptedException exp) {
-                Thread.currentThread().interrupt();
                 extor.shutdownNow();
             }
         }
     }
+
+    /**
+     * Execute the tasks concurrently using {@link Threads} when
+     *  {@link ExecutorService} was initiated by someone else.
+     */
+    @Test
+    public void cactoosWay() {
+        this.repeatAtLeastTenTimesWithTimeoutInFiveSeconds(
+            arg -> {
+                final ExecutorService extor = Executors.newFixedThreadPool(3);
+                try {
+                    MatcherAssert.assertThat(
+                        new Threads<String>(
+                            extor,
+                            () -> {
+                                TimeUnit.SECONDS.sleep(3);
+                                return "txt 1";
+                            },
+                            () -> {
+                                TimeUnit.SECONDS.sleep(3);
+                                return "txt 2";
+                            },
+                            () -> {
+                                TimeUnit.SECONDS.sleep(3);
+                                return "txt 3";
+                            }
+                        ),
+                        new HasValues<>("txt 1", "txt 2", "txt 3")
+                    );
+                } finally {
+                    try {
+                        extor.shutdown();
+                        if (!extor.awaitTermination(1, TimeUnit.SECONDS)) {
+                            extor.shutdownNow();
+                        }
+                    } catch (final InterruptedException exp) {
+                        extor.shutdownNow();
+                    }
+                }
+            }
+        );
+    }
+
+    /**
+     * Execute the test at least 10 times with timeout in 5 seconds each.
+     * @param test The test to execute.
+     * @todo #972:30min Move this method to new object RepeatWithTimeout.
```

Comment:
@dgroup I don't agree with this puzzle. We already have Repeat and Timeout functionalities. You can just compose our primitives to achieve "repeat with timeout", as you've done here.